src/ir/types/type.h

```diff
@@ -19,6 +19,24 @@ class Type {
   virtual Kind kind() const = 0;
 };
 
+
+class Type {
+ public:
+  Type(std::unique_ptr<TypeBase> type) : type_(std::move(type)) {}
+
+  raksha::ir::AccessPathSelectorsSet GetAccessPathSelectorsSet() const {
+    return type_->GetAccessPathSelectorsSet();
+  }
+
+  // TODO: This should be removed to hide the TypeBase.  method is added as a
+  // stop-gap solution to work with the rest of the code base.
+  const TypeBase& type_base() const { return *type_.get(); }
+
+ private:
+  std::unique_ptr<TypeBase> type_;
+};
```

Comment:
I wonder about this approach. I was thinking that `TypeBase` would be paired with something like `TypeRef`, where `TypeRef` would contain within it a small, efficiently-copyable unique ID that could be used to look up a `TypeBase` in a map in the factory. This `TypeRef` could then be used in multiple places without running into uniqueness problems, while the `std::unique_ptr` would be held by the factory.

I'm also concerned about whether or not we're helping by wrapping the `std::unique_ptr` in this fashion. Certainly `Type` is more concise than `std::unique_ptr<TypeBase>`, but it does hide the fact that the type cannot be copied and that it implies ownership.

Reply:
This is just an intermediate step to unblock the rest of the IR work. Having something like a copyable `TypeRef` is the plan. In fact, `Type` would be the `TypeRef` you are referring to. However, this required fleshing out the canonicalization of types. So, I just started by creating the wrapper interface so that we can use it in the `operator`, `operation`, etc.